alibi/explainers/anchor_image.py

```diff
@@ -344,9 +345,11 @@ def __init__(self,
         super().__init__(meta=copy.deepcopy(DEFAULT_META_ANCHOR))
         np.random.seed(seed)
 
-        if isinstance(segmentation_fn, str) and not segmentation_kwargs:
+        # TODO: this logic needs improvement. We should check against a fixed set of strings
+        # for built-ins instead of any `str`.
+        if isinstance(segmentation_fn, str) and segmentation_kwargs is None:
```

Comment:
`and not` was a bug as it would also match the empty dictionary. Now we explicitly deal with the `None` case which conveniently doesn't require the `type: ignore` later on.